polymath/extensions/indexer.py

```
@@ -4,588 +4,592 @@
+            # Simplify the post_mask if possible
+            if not all(array_shape):        # mask doesn't matter if size is zero
+                post_mask = False
+            elif np.all(post_mask):
+                post_mask = True
 
-    except Exception as e:
-        raise IndexError(e)
+            return (tuple(pre_index), post_mask, has_ellipsis, moved_to_front, array_shape,
+                    first_array_loc)
 
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)